### PR TITLE
Port chmlib and chm_extra plugins to build on python2/python3

### PR DESCRIPTION
**Pull request by eli-schwartz**

### CODE REVIEW

File: src/calibre/utils/chm/extra.c [hunk 1]
```diff
@@ -42,6 +51,13 @@
 #                       define MODEXPORT(a) __declspec(dllexport) a
 #                       define MODIMPORT(a) extern a
 #               endif
+#define uint64_t unsigned long long
+#define uint32_t unsigned int
+#define uint16_t unsigned short
+#define uint8_t  unsigned char
+#define size_t   int
```
Comment (kovidgoyal) [inline src/calibre/utils/chm/extra.c]: this looks wrong. as of VS 2015 (which calibre uses on windows, all these types should be defined anyway, so no need to redefine them here.

Comment (eli-schwartz) [reply src/calibre/utils/chm/extra.c]: It is automagically generated by swig, probably under the assumption that people use very old compilers.

**Comment by kovidgoyal** [comment]
Also, I seriously doubt this will actually work on py3, the python bits as oppossed to the c bits are going to require refactoring most likely, because they currently work on bytes and the python 3 bytes type is much less capable, so if anybody does anything like slicing or regexes, thing are likely to break.

File: src/calibre/utils/chm/extra.c (hunk 1 above)
Comment (eli-schwartz) [reply src/calibre/utils/chm/extra.c]: Perhaps https://github.com/dottedmag/pychm/pull/5 is meant to work on very old compilers.

Comment (kovidgoyal) [reply src/calibre/utils/chm/extra.c]: yeah but even then size_t is an unsigned type while int is signed. That cant be right.

**Comment by kovidgoyal** [comment]
I decided to modernize the SWIG wrapper for chmlib. It now has no hand written C code an no swig generated python code, so it should be buildable on python 3 automatically.

**Pull request closed without merging**